source/restrict_responses.md

```diff
@@ -1,104 +1,105 @@
-* Restrict an input to a non-digit character (e.g. a or c or ! or # or % etc.): **regex(., '^\D$')**
-* Restrict an input to five non-digit character (e.g. **aZcB!**#% etc.): **regex(., '^\D{5 }$')**
-
-#### _Examples related to use of a combination of alphabets letters and numbers_
-
-* Restrict one character which matches between a to z or A to Z or 0 to 9 or _ (i.e. match one character from **[a-zA-Z0-9_]**): **regex(., '^\w$')**
-* Restrict three character which matches between a to z or A to Z or 0 to 9 or _ (i.e. match one character from **[a-zA-Z0-9_]**): **regex(., '^\w{3}$')**
-* Restrict your beneficiary ID to a specific format e.g. **CAR_PRC_2020_0048**: **regex(., '^[A-Z]{3}[_][A-Z]{3}[_][0-9]{4}[_][0-9]{4}$')**
-* Restrict your beneficiary ID to a specific format e.g. **CAR-PRC-2020-0048** (_where the enumerators should enter an exact match from **CAR** to - i.e. **CAR-PRC-2020-**_ and can enter any 4 digit serial number): **regex(., '^CAR-PRC-2020-[0-9]{4}$')**
-* Restrict a currency input of three digits with a currency sign (either dollar or pound) in front (e.g. **$999** or **£500**): **regex(., '^[\$|\£]\d{3}$')**
-* Restrict an exact input of number of words (e.g. to restrict exactly 3 words "_I love you._"): **regex(., '^\W*(\w+\b\W*){3}$')**
-* Restrict an input of number of words (e.g. to restrict a range of words say 3 to 5): **regex(., '^\W*(\w+\b\W*){3,5}$')**
+* Restrict an input of the beneficiaries name where the initials of the first name and last name are uppercase e.g. _Kobe Bryant_: **regex(., '^[A-Z]{1}[a-z]{1,}[ ]{1}[A-Z]{1}[a-z]{1,}$')**
+* Restrict an input of the beneficiaries name with first name, middle name (if any) and last name e.g. _Kobe Bean Bryant_: **regex(., '^\w{1,}\s(\w{1,})?(\s)?\w{1,}$')**
+* Restrict an input of the beneficiaries' full name where the initials of the names are in uppercase and the name are quite long (often greater than 3 words) e.g. _Samayamantri Venkata Rama Naga Butchi Anjaneya Satya Krishna Vijay_ (this is an example of south Indian names): **regex(., '^([A-Z]{1}[a-z]{1,}\s)([A-Z]{1}[a-z]{1,}\s?)+$')**
+* Restrict an input of the beneficiaries' first name (so that you are able to capture the exact spelling) where the enumerators are forced to enter the beneficiaries first name twice e.g. _Kobe Bryant Kobe_. (This could be helpful when you are trying to document beneficiaries details where a typo error could cost you heavy): **regex(., '^(\D+)\s(\D+)\s?\1$')**
+* Restrict an input of the beneficiaries' last name (so that you are able to capture the exact spelling) where the enumerators are forced to enter the beneficiaries last name twice e.g. _Kobe Bryant Bryant_. _(This could be helpful when you are trying to document beneficiaries details where a typo error could cost you heavy)_: **regex(., '^(\D+)\s(\D+)\s?\2$')**
+* Restrict a character within a word by using the _?_ (quantifier) e.g. allow either _color_ or _colour_ as an input: **regex(., '^colou?r$')**
```

Comment:
In addition to the above, rather use back ticks for regex-specific symbols too: i.e _?_ becomes `?`

Reply:
@Kalyan-Lama hey 👋